### DISCUSSION TITLE
How to connect Kafka cluster with Kafka tool

### DISCUSSION THREAD

**Discussion by jiangjian0920** [strimzi/strimzi-kafka-operator#4285] [General]
Kafka tool need the zookeeper host and port ，how to expose that。

**Comment by scholzj** [comment]
We do not support exposing Zookeeper. This is historically because Zookeeper did not support proper security in the past and because now - when it supports security such as TLS - it will be removed soon from Kafka. Most Kafka tools do not need to connect to Zookeeper and can be used against Kafka directly. If you really need to use Zookeeper, you can:
* Exec into one of the Zookeeper pods and connect to `localhost:12181`
* Use the Zoo entrance (https://github.com/scholzj/zoo-entrance) which opens an unencrypted and unsecured port with which you can access Zookeeper. (This is not official part of the Strimzi project. Use at your own risk.)
